test/assertions_test.rb

```diff
@@ -124,7 +124,7 @@ def test_assert_statsd_call
       end
     end
 
-    assert_assertion_triggered do
+    assert_no_assertion_triggered do
```

Comment:
I think the simple syntax should cater for the most common use case.

If we want to support more advanced use cases, we can make the `MetricExpectation` class have a more advanced "matcher" API (e.g. for assertion that a tag is absent). Users can construct instances an send them to `assert_statsd_calls`. Let's only cross that bridge when there's a need for it though.